pkgs/development/compilers/halvm/2.4.0.nix

```diff
@@ -20,9 +20,9 @@ stdenv.mkDerivation rec {
     sed -ie 's|ld |${targetPackages.stdenv.cc.bintools}/bin/ld |g' src/scripts/ldkernel.in
   '';
   configureFlags = stdenv.lib.optional (!enableIntegerSimple) [ "--enable-gmp" ];
-  propagatedNativeBuildInputs = [ alex happy ];
+  propagatedNativeBuildInputs = [ (stdenv.lib.getBin alex) (stdenv.lib.getBin happy) ];
```

Comment:
My understanding is that a reference like `alex` refers to the output by default that's defined *first*. Is that correct?

If that is so indeed, then we should make sure that the first output defined by Haskell packages is the one containing the binaries so that these references don't need to be wrapped in `getBin` all the time.

Reply:
`bin` is the first one, this code is just trying to be explicit.

See https://github.com/NixOS/nixpkgs/blob/haskell-outputs-bin-etc-lib/pkgs/development/haskell-modules/generic-builder.nix#L204-L212

Reply:
I think that being explicit in this case is not an advantage since it makes the code harder to read without adding any functional benefit. People might be confused by those explicit calls, wondering: "Why is this necessary?" Personally, I'd be in favor of dropping `getBin` calls whereever possible. Rather than being explicit, we should document that `bin` can be relied upon to be the first `output` of all Haskell derivations.

Reply:
It doesn't need to be specifically documented for Haskell since it should hold for all of nixpkgs: https://github.com/NixOS/nixpkgs/blob/master/doc/multiple-output.xml#L39